src/hooks/tests/useDimensions.test.ts

```diff
@@ -0,0 +1,133 @@
+/* eslint-disable class-methods-use-this */
```

Comment:
Check out the tests for `useIntersectionObserver` they behave similarly 
https://github.com/evergreen-ci/spruce/blob/c4b2c563e1f49d71e3a509122c594568025aa65a/src/hooks/useIntersectionObserver/useIntersectionObserver.test.ts